src/webui/www/public/scripts/login.js

```diff
@@ -28,10 +28,63 @@
 
 'use strict';
 
-document.addEventListener('DOMContentLoaded', function() {
-    document.getElementById('username').focus();
-    document.getElementById('username').select();
-});
+async function setupI18n() {
+    const languages = (() => {
+        const langs = new Set();
+        for (const lang of navigator.languages) {
+            langs.add(lang.replace('-', '_'));
+
+            const idx = lang.indexOf('-');
+            if (idx > 0)
+                langs.add(lang.slice(0, idx));
+        }
+        langs.add('en'); // fallback
+        return Array.from(langs);
+    })();
+
+    // it is faster to fetch all translation files at once than one by one
+    const fetches = languages.map(lang => fetch(`lang/${lang}.json`));
+    const fetchResults = await Promise.allSettled(fetches);
+    const translations = fetchResults
+        .map((value, idx) => ({ lang: languages[idx], result: value }))
+        .filter(v => (v.result.value.status === 200));
+    const translation = {
+        lang: translations[0].lang.replace('_', '-'),
+        data: await translations[0].result.value.json()
+    };
+
+    // present it to i18next
+    const i18nextOptions = {
```

Comment:
> Since we will download all the languages anyway

Is this really the case? It looks unreasonable. Is it possible to avoid this? Or are we talking about the names of languages, and not about translations? (I'm sorry, I don't go into its details.)

Reply:
>Is this really the case? It looks unreasonable.

There's a code comment a few lines above. The rationale is it is faster to fetch them all together vs one by one.

IMO, in the case of the "webroot/public/" portion of the WebUI it is ok to do it that way. The translatable strings are very very few.
For the "webroot/private/" portion we would probably need to download them on demand. There is probably a [backend/plugin](https://www.i18next.com/overview/plugins-and-utils#backends) that can help with that. But it is probably best to discuss it in the PR that will implement it for the "webroot/private/" portion.

Reply:
>But we only need single one, don't we?

We can't know if the user preferred translation exists in qbt or not so we fetch all possible matches and present the closest match.
Also fetching them one by one is too slow in case the transmission delay is large.

>Since we will download all the languages anyway, give them all to i18next. Then control the presented langauge via the lng key and the fallback language (aka en) via the fallbackLng key.

IMO there is no point. Feeding more than what i18next needs doesn't improve anything. From my testing it will only use the major language and only when it doesn't exist then the fallback lang will be used.

Reply:
>For the "webroot/private/" portion we would probably need to download them on demand. There is probably a [backend/plugin](https://www.i18next.com/overview/plugins-and-utils#backends) that can help with that. But it is probably best to discuss it in the PR that will implement it for the "webroot/private/" portion.

I would still do it the same way as public folder unless the translation resource is > 1MB large. Besides, the translation resource will be compressed by the server so the actual data over the wire should be way smaller.